pilot/pkg/ambient/ambientpod/podutil.go

```diff
@@ -51,10 +51,9 @@ func ShouldPodBeInIpset(namespace *corev1.Namespace, pod *corev1.Pod, meshMode s
 	return false
 }
 
-// @TODO Interim function for waypoint proxy, to be replaced after design meeting
 func PodHasOptOut(pod *corev1.Pod) bool {
-	if val, ok := pod.Labels["ambient-type"]; ok {
-		return val == "waypoint" || val == "none"
+	if val, ok := pod.Annotations[constants.AmbientRedirection]; ok {
+		return val == constants.AmbientRedirectionDisabled
```

Comment:
So this is excluding pod that is explicitly disabled by user

Reply:
Yes